### PR TITLE
Create "Addons By Name" table

### PR DESCRIPTION
**Pull request by DNin01**
Partially addresses #5674

### Changes

Adds a new "Addons By Name" document as `/addons/README.md`. It's a table of addon names (in alphabetical order) along with their IDs.

The table is automatically generated via the `gen-addon-table.py` Python script. Every time new addons are added or addon names are changed, this script must be re-run to update the addon table.

I'm new to Python, so let me know if there's something I can improve!

I'm also not sure if we need this, but I'll leave this open.

### Reason for changes

It can be hard to find the addon IDs of certain addons because sometimes they do not closely correspond to their addon's name or are ambiguous. The table solves that problem. It's also useful if you want to reference an addon by its ID but don't know it.

### Tests

Tested on Windows with Python 3.11.

### CODE REVIEW

File: addons/README.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
+# Addons By Name
```
Comment (DNin01) [inline addons/README.md:1]: Should "By" be uppercase or lowercase? I couldn't decide.

File: dev/gen-addon-table.py
Comment (DNin01) [inline dev/gen-addon-table.py:1]: Is the `dev` directory a good place for this script?

Comment (DNin01) [inline dev/gen-addon-table.py:1]: Not sure if we should have some forms of sanitization or directory checking (making sure we don't go outside the intended directories).

**Comment by DNin01** [comment]
I wonder if a script like this could be used to build the settings page's addon list (so our extension doesn't have to repeatedly spend time generating it), or if it could generate a single file containing all the addon manifests in one file (would that be useful at all?).

Requiring contributors to run a script could raise the floor for contributors, though. But more software is starting to use Node.js on the development side, so...

**Comment by Joeclinton1** [comment]
Disapprove. The best way to handle this is a util function that takes an addon name and returns an ID. This function would perform the same internal logic as this addon, but would mean there is no need for a static file that needs to be mantained and updated. There should not be two sources of truth.

**Comment by DNin01** [comment]
I was going to say, why not just make a page part of the Scratch Addons extension that does the same thing?

I don't really like unnecessarily complicating the development process.

**Comment by DNin01** [comment]
But you can also just hover over addon names to see their IDs in the settings page (#6786), so this is already sort of solved?

**Comment by WorldLanguages** [comment]
Yeah, this is a quite similar concept to https://scratchaddons.com/addons/.

**Pull request closed without merging**